src/components/App.tsx

```diff
@@ -1,3 +1,85 @@
+import { useState, useEffect } from "react"
+import { Input } from "./common/input"
+import { Button } from "./common/button"
+import { Checkbox } from "./common/checkbox"
+import { Radiogroup } from "./common/radiogroup"
+import css from "./styles.module.css"
+
+
+
+
 export const App = () => {
-    return (<div></div>)
+
+    const [inputValue, setInputValue] = useState("")
+    const [tasks, setTasks] = useState<{ id: number, isDone: boolean, value: string }[]>([{ id: 1, isDone: false, value: "Выучить реакт" }])
+    const [filter, setFilter] = useState("all")
+
+
+    useEffect(() => {
+        const tasks = JSON.parse(localStorage.getItem("tasks") ?? '[]');
+        if (tasks.length) {
+            setTasks(tasks)
+        }
+    }, [])
+
+    useEffect(() => {
+        localStorage.setItem("tasks", JSON.stringify(tasks));
+    }, [tasks])
+
+    const addTaskHandler = () => {
+        if (inputValue.trim().length === 0) {
+            return alert('Не валидное имя задачи');
+        }
+        setTasks([...tasks, { id: tasks.length + 1, isDone: false, value: inputValue }]);
```

Comment:
тут рассчитываем стейт на основе предыдущего (добавляем таску в предыдущий массив), поэтому нужно использовать синтаксис колбека
```suggestion
        setTasks((prevTasks) => [...prevTasks, { id: tasks.length + 1, isDone: false, value: inputValue }]);
```

id на основе длинны массива приведет к багам (например если удалить таску из середины и потом добавить, то у тебя будет две таски с одним id).
В примере что я кидал есть либа попробуй ее установить `npm i uuidv4` и использовать.